File: server/src/main/java/rs/ac/singidunum/novisad/server/model/student/StudentskaAktivnost.java
package rs.ac.singidunum.novisad.server.model.student;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.ac.singidunum.novisad.server.model.predmet.Predmet;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class StudentskaAktivnost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    private Aktivnost aktivnost;
    @OneToOne
    private StudentNaGodini student;
    @OneToOne
    private Predmet predmet;
    private Double osvojenoPoena;
    private LocalDate datum;

}

File: client/src/app/model/studentskaAktivnost.ts
import {Aktivnost} from "./aktivnost";
import {StudentNaGodini} from "./studentNaGodini";
import {Predmet} from "./predmet";

export interface StudentskaAktivnost {
  id: number,
  aktivnost: Aktivnost,
  student: StudentNaGodini,
  predmet: Predmet,
  osvojenoPoena: number,
  datum: Date
}
